### PR TITLE
feat(version): bump prerelease versions from conventional commits (#2…

### PR DESCRIPTION
**Pull request by amorscher**
Adds possibility to force a version bump, even if already existing version is a prerelease version. This is achieved by introducing a new flag for `lerna version`  command named `--conventional-bumpprerelease`.

## Description
Adds new flag `--conventional-bumpprerelease` to force a version bump, if required according to conventional commits. When run with this flag, `lerna version` will release with bumped prerelease versions, even if already released packages are prereleases. Releases all unreleased changes as pre(patch/minor/major/release) by prefixing the version recommendation from `conventional-commits` with `pre`, eg. if present changes include a feature commit, the recommended bump will be `minor`, so this flag will result in a `preminor` release. If not used just a prerelease bump will be applied to prereleased packages.

## Motivation and Context
Currently, if the existing version is a prerelease no version bump is performed, although there exists conventional commits which would require that. Following version bumps are performed using `--conventional-bumpprerelease`:
```sh
Changes:
 - major: 1.0.0-alpha.0 => 2.0.0-alpha.0 
 - minor: 1.0.0-alpha.0 => 1.1.0-alpha.0 
 - patch: 1.0.0-alpha.0 => 1.0.1-alpha.0
``` 
Behaviour without using the flag is:
```sh
Changes:
 - major: 1.0.0-alpha.0 => 1.0.0-alpha.1 
 - minor: 1.0.0-alpha.0 => 1.0.0-alpha.1 
 - patch: 1.0.0-alpha.0 => 1.0.0-alpha.1
``` 
The issue/question is described here (#2536).

## How Has This Been Tested?
Added several unit tests and tested manually with locally published lerna version.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (change that has absolutely no effect on users)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by amorscher** [comment]
What can I do that someone takes a look at this pull request.

Is there any additional information needed? Some feedback would be nice.

What is more it would be cool if I get some authorization to run the required workflows. How can I achieve this?

 Thx a lot!

**Comment by JamesHenry** [comment]
@amorscher Thanks a lot for your patience on this! I am currently recovering from COVID while traveling internationally, so am taking a break at the moment. I will discuss this potential feature with the team towards the end of next week. The workflows do not run by default for folks who have never contributed before. I have enabled them now but it looks like in this case it failed the first check. I recommend getting the checks we specify in `ci.yml` to pass for you locally prior to pushing to help reduce the iteration time in CI. Many thanks again for your contribution!

**Comment by amorscher** [comment]
@JamesHenry THX for the response and for bringing it to the team. I updated the pull request. Now the workflows should pass. I ran them locally, despite some minor snapshot differences due to committer id and CPU infos all of them passed. 

Wish you a good recorvery! I really know that takes some time.

**Comment by amorscher** [comment]
@JamesHenry Looks like I need another approval for running the workflows. Is this all the time needed for the first contribution?

File: commands/version/README.md
Comment (JamesHenry) [inline commands/version/README.md:205]: Let's please go for `--conventional-bump-prerelease`, I don't see any reason to treat `bumpprerelease` as one word

File: commands/version/command.js
Comment (JamesHenry) [inline commands/version/command.js:36]:
```suggestion
      describe: "Bumps prerelease versions if conventional commits requires it.",
```

File: commands/version/index.js
Comment (JamesHenry) [inline commands/version/index.js:370]: See note on not treating `bumpprerelease` as one word

File: core/conventional-commits/lib/recommend-version.js
Comment (JamesHenry) [inline core/conventional-commits/lib/recommend-version.js:33]: Given this function's purpose is to dynamically determine whether or not to bump, it feels strange to invoke it with a ready made answer... I think let's instead refactor line 62 (what is currently line 66 in this diff)

File: core/conventional-commits/lib/recommend-version.js
Comment (JamesHenry) [inline core/conventional-commits/lib/recommend-version.js:66]:
```suggestion
          const shouldBump = conventionalBumpprerelease || shouldBumpPrerelease(releaseType, pkg.version);
```

File: core/conventional-commits/lib/recommend-version.js
Comment (JamesHenry) [inline core/conventional-commits/lib/recommend-version.js:18]: See note on not treating `bumpprerelease` as one word

**Review by JamesHenry — CHANGES_REQUESTED** [review]
Thanks a lot @amorscher, I'm doing much better now and we agreed at the end of this week to press ahead with this feature!

Please see my various items of feedback inline.

Please could you also add an e2e test to cover the use of this flag?

**Comment by ghiscoding** [comment]
another thing missing in this PR is the [lerna-schema.json](https://github.com/lerna/lerna/blob/main/core/lerna/schemas/lerna-schema.json) that needs to be updated with the new option

**Comment by amorscher** [comment]
> Thanks a lot @amorscher, I'm doing much better now and we agreed at the end of this week to press ahead with this feature!
> 
> Please see my various items of feedback inline.
> 
> Please could you also add an e2e test to cover the use of this flag?

Thanks a lot for the review. I am really happy that the feature is pressed ahead.

I integrated requested changes and I will add an e2e test during the week.

**Review by JamesHenry — APPROVED** [review]

**Comment by JamesHenry** [comment]
Thank you @amorscher!

**Pull request merged**
